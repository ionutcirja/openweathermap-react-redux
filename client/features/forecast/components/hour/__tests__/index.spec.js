import React from 'react';
import { shallow } from 'enzyme';
import { Component } from '..';

describe('ForecastHour component', () => {
  const propsToRender = {
    hour: '12:00',
    icon: 'icon',
    description: 'info',
    temp: 100,
    humidity: 200,
    pressure: 1000,
    theme: {
      colours: {
        grey: 'grey',
        greyLight: 'grey-light',
      },
    },
  };

  describe('render', () => {
    it('should render the hour, an icon representing the weather condition'
      + ' and temperature, humidity and pressure values', () => {
      const wrapper = shallow(<Component {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
