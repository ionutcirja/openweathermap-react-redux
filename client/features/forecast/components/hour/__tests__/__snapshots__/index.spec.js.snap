// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastHour component render should render the hour, an icon representing the weather condition and temperature, humidity and pressure values 1`] = `
<Hour>
  <Title
    colour="grey"
  >
    12:00
  </Title>
  <img
    alt="info"
    src="http://openweathermap.org/img/w/icon.png"
  />
  <p>
    <Details
      colour="grey"
    >
      Temperature: 100
    </Details>
    <Details
      colour="grey"
    >
      Humidity: 200
    </Details>
    <Details
      colour="grey"
    >
      Pressure: 1000
    </Details>
  </p>
</Hour>
`;
